ChiModules/DiffusionSolver/chi_diffusion_utils.cc

```diff
@@ -25,11 +25,13 @@ KSPMonitorAChiTech(KSP ksp, PetscInt n, PetscReal rnorm, void *monitordestroy)
     char buff[100];
     if (rnorm/rhs_norm < 1.0e-2)
     {
-      snprintf(buff,100,"Diffusion iteration %4lld - Residual %.3e\n",n,rnorm/rhs_norm);
+      snprintf(buff,100,"Diffusion iteration %4ld - Residual %.3e\n",
+               n,rnorm/rhs_norm);
```

Comment:
This likely doesn't matter much, but I get compiler warnings such as:

`warning: format '%ld' expects argument of type 'long int', but argument 4 has type 'PetscInt' {aka 'long long int'} [-Wformat=]`

with GCC-11.1.0 and PETSc-3.15.1

Reply:
I observed the opposite issue when compiling on various different machines. Anyway, I have decided to transition these parts of the code to C++-style streams, which should avoid system-/implementation-dependent issues altogether.